### PR TITLE
First take on `flutter start` for iOS devices

### PR DESCRIPTION
**Pull request by chinmaygarde**

### CODE REVIEW

File: packages/flutter_tools/lib/src/device.dart [hunk 1]
```diff
@@ -337,8 +351,6 @@ class IOSSimulator extends Device {
     List<IOSSimulator> devices = [];
     String id = _getRunningSimulatorID(mockIOS);
     if (id != null) {
```
Comment (Hixie) [inline packages/flutter_tools/lib/src/device.dart]: no braces around one-line if statements

**Comment by Hixie** [comment]
lgtm.

I think we should refactor the code so that we don't need any of these switches based on platform, though. All the platform-specific code should be in platform-specific classes, and the main code should just drive it without knowledge of the differences.

**Comment by chinmaygarde** [comment]
@Hixie: I am landing this now. @devoncarew is also planning to refactor `device.dart`. Will pick this up again after.

**Comment by Hixie** [comment]
Oh yeah, totally. I didn't mean for that comment to prevent check-in. Just "directions for future research".

**Pull request merged**